File: python/7_kyu/gcd_sum.py
# GCD sum
# https://www.codewars.com/kata/5dd259444228280032b1ed2a/train/python

def solve(s,g):
    results = []
    
    if s % g == 0 or g % s == 0:

      for i in range(g, s):
        if i % g == 0:
          results.append(i)
          break
      
      for i in range(g, s):
        if i + results[0] == s:
          results.append(i)
          break

      return tuple(results)
    
    else:
      return -1

# Best Solution
def solve(s,g):
    return -1 if s % g else (g, s - g)

File: javascript/7_kyu/gcd_sum.js
// GCD sum
// https://www.codewars.com/kata/5dd259444228280032b1ed2a/train/javascript

function solve(s,g) {
    var ans = [];

    if ( (s % g == 0) || (g % s == 0) ){
        for (var i = g; i < s; i++) {
            if (i % g == 0) {
                ans.push(i);
                break;
            }
        }
    
        for (var i = g; i < s; i++) {
            if (i + ans[0] == s) {
                ans.push(i);
                break;
            }
        }
        return ans;

    } else {
        return -1;
    }
}

console.log(solve(6,3));

// Best Answer
// function solve(x, y) {
//     return x % y ? -1 : [y, x - y];
//   }